elyra/pipeline/airflow/component_parser_airflow.py

```diff
@@ -46,149 +39,316 @@ class AirflowComponentParser(ComponentParser):
+                # Without the init function, class can't be parsed for properties
+                # TODO is this the correct functionality? see classes that derive from PythonOperator
+                self.log.warning(f"Operator '{component_class}' associated with identifier "
+                                 f"'{registry_entry.component_identifier}' does not have an __init__ "
+                                 f"function. Skipping...")
+                continue
 
-        for component_class, component_content in component_classes.items():
-            # Create a Component object for each class
-            component_properties = self._parse_properties(component_content)
+            # Assign component name and unique id
+            component_id = registry_entry.component_id
+            if num_operator_classes > 1:
+                # This file contains more than one operator and id must be adjusted
+                # to include the Operator class name as well
+                component_id += f":{component_class}"
+
+            # Get the properties for this Operator class
+            try:
+                component_properties: List[ComponentParameter] = self._parse_properties_from_init(**content)
+            except Exception as e:
+                self.log.error(f"Properties of operator '{component_class}' associated with "
+                               f"identifier '{registry_entry.component_identifier}' could not be "
+                               f"parsed: {e}. Skipping...")
+                continue
+            if not component_properties:
+                self.log.warning(f"Operator '{component_class}' associated with identifier "
+                                 f"'{registry_entry.component_identifier}' does not have any properties. "
+                                 f"Skipping...")
+                continue
 
-            new_component = Component(id=registry_entry.component_id,
-                                      name=component_class,
-                                      description='',
-                                      catalog_type=registry_entry.catalog_type,
-                                      source_identifier=registry_entry.component_identifier,
-                                      definition=self.get_class_def_as_string(component_content),
-                                      runtime_type=self.component_platform.name,
-                                      categories=registry_entry.categories,
-                                      properties=component_properties
-                                      )
+            new_component = Component(
+                id=component_id,
+                name=component_class,
+                description=DEFAULT_DESCRIPTION,
+                catalog_type=registry_entry.catalog_type,
+                source_identifier=registry_entry.component_identifier,
+                definition=component_definition,  # TODO use full file contents or only the class contents?
```

Comment:
Related to the to-do: not sure it really matters either way since this field isn't currently used for Airflow (it is for KFP)